joblib/test/test_memory.py

```diff
@@ -229,8 +229,8 @@ def name_collision(x):
         a(1)
         b(1)
 
-        yield assert_equal, len(w), 1
-        yield assert_true, "collision" in str(w[-1].message)
+        assert len(w) == 1
```

Comment:
Sometimes `yield` is useful to have multiple smaller tests in the same test function and keep going even executing the function if there is a failure. I am not sure whether there is a py.test equivalent for this.

For example in this particular example it could well be that I have 2 warnings but the last one contains 'collision'. In this case using yield, I will have the first assert failing but the second passing. With normal asserts the first assert failing will stop the execution and I won't know whether the second fails or not.